src/beamme/four_c/model_importer.py

```diff
@@ -106,177 +100,123 @@ def import_four_c_model(
-    ](nodes=nodes, data=element["data"])
+    # function to pop sections from the input file
+    _pop_section = lambda name: input_file.pop(name, [])
 
-    # Check if we have to link this element to a material object (rigid spheres do not
-    # have a material).
-    if "MAT" in created_element.data:
-        created_element.data["MAT"] = material_id_map[created_element.data["MAT"]]
-    return created_element
+    # convert all sections to native objects and add to a new mesh
+    mesh = _Mesh()
 
+    # extract materials
+    material_id_map = {}
 
-def _boundary_condition_from_dict(
-    geometry_set: _GeometrySetNodes,
-    bc_key: _Union[_conf.BoundaryCondition, str],
-    data: _Dict,
-) -> _BoundaryConditionBase:
-    """This function acts as a factory and creates the correct boundary
-    condition object from a dictionary parsed from an input file."""
+    for mat in _pop_section("MATERIALS"):
+        mat_id = mat.pop("MAT")
+        if len(mat) != 1:
+            raise ValueError(
+                f"Could not convert the material data `{mat}` to a BeamMe material!"
+            )
+        mat_name, mat_data = next(iter(mat.items()))
+        material = _MaterialSolid(material_string=mat_name, data=mat_data)
+        mesh.add(material)
+        material_id_map[mat_id] = material
+
+    # extract nodes
+    mesh.nodes = [_Node(node["COORD"]) for node in _pop_section("NODE COORDS")]
+
+    # extract elements
+    for input_element in _pop_section("STRUCTURE ELEMENTS"):
+        if (
+            input_element["cell"]["type"]
+            not in _INPUT_FILE_MAPPINGS["element_four_c_string_to_type"]
+        ):
+            raise TypeError(
+                f"Could not create a BeamMe element for `{input_element['data']['type']}` `{input_element['cell']['type']}`!"
+            )
+        nodes = [mesh.nodes[i - 1] for i in input_element["cell"]["connectivity"]]
+        element_class = _INPUT_FILE_MAPPINGS["element_four_c_string_to_type"][
+            input_element["cell"]["type"]
+        ]
+        element = element_class(nodes=nodes, data=input_element["data"])
+        if "MAT" in element.data:
+            element.data["MAT"] = material_id_map[element.data["MAT"]]
+        mesh.elements.append(element)
+
+    # extract geometry sets
+    geometry_sets_in_sections: dict[str, dict[int, _GeometrySetNodes]] = {
+        key: {} for key in _bme.geo
+    }
 
-    del data["E"]
+    for section_name in input_file.sections:
+        if not section_name.endswith("TOPOLOGY"):
+            continue
+
+        items = _pop_section(section_name)
+        if not items:
+            continue
+
+        # Find geometry key for this section
+        geometry_key = None
+        for key, val in _INPUT_FILE_MAPPINGS[
+            "geometry_sets_geometry_to_condition_name"
+        ].items():
+            if val == section_name:
+                geometry_key = key
+                break
+        if geometry_key is None:
```

Comment:
```suggestion
        else:
```
I think that should work.